### ISSUE TITLE
OneCycleLR scheduler looks wrong

### ISSUE THREAD

**Issue by LearnedVector** [PyTorchLightning/pytorch-lightning#3205]
## 🐛 Bug

I'm using the `optim.lr_scheduler.OneCycleLR` and it seems to be cycling up and down every 1k iteration. anybody understand why? here is my implementation...

```
    def configure_optimizers(self):
        self.optimizer = optim.AdamW(self.model.parameters(), self.args.learning_rate)

        ds_len = self.train_dataset.__len__()
        total_bs = self.args.batch_size*self.args.gpus
        self.scheduler = optim.lr_scheduler.OneCycleLR(
                                        self.optimizer, max_lr=self.args.learning_rate,
                                        anneal_strategy='linear', div_factor=100,
                                        steps_per_epoch=int((ds_len/total_bs)),
                                        epochs=self.args.epochs)

        sched = {
            'scheduler': self.scheduler,
            'interval': 'step',
        }
        return [self.optimizer], [sched]
``` 

And here is what it looks like...

![Screen Shot 2020-08-26 at 3 06 43 PM](https://user-images.githubusercontent.com/8495552/91351509-0906fc80-e7ae-11ea-9de1-4800cec85f5c.png)

### To Reproduce

1. use the optimizer configuration code above
2. use a tensorboard logger
3. in training step, log the learning rate to tensorboard
```
    def training_step(self, batch, batch_idx):
        loss = self.step(batch)
        lr = self.scheduler.get_last_lr()
        logs = {'loss': loss, 'lr': lr[0].detach().cpu().numpy()}
        return {'loss': loss, 'log': logs}
```

### Expected behavior

The one cycle learning rate should be a single cycle where it goes up and down. Not sure what is happening here.

### Environment

Please copy and paste the output from our
[environment collection script](https://raw.githubusercontent.com/PyTorchLightning/pytorch-lightning/master/tests/collect_env_details.py)
(or fill out the checklist below manually).

You can get the script and run it with:
```
wget https://raw.githubusercontent.com/PyTorchLightning/pytorch-lightning/master/tests/collect_env_details.py
# For security purposes, please check the contents of collect_env_details.py before running it.
python collect_env_details.py
```

 - PyTorch Version (e.g., 1.0): 1.6
 - OS (e.g., Linux): Linux
 - How you installed PyTorch (`conda`, `pip`, source): pip
 - Build command you used (if compiling from source):
 - Python version: 3.7.0
 - CUDA/cuDNN version: 10.2
 - GPU models and configuration: GeForce RTX 2080 Ti

**Comment by ananyahjha93** [comment]
@LearnedVector are you using DDP or DP for training? Does the same issue occur if you provide ```total_steps``` param instead?

**Comment by LearnedVector** [comment]
@ananyahjha93 I am using `ddp`. and yes if i provide the `total_steps` same issue is occurring!

**Comment by ananyahjha93** [comment]
@LearnedVector can you print out these values in the code and provide them here.
- ds_len
- batch_size
- gpu count
- epoch count

**Comment by LearnedVector** [comment]
@ananyahjha93  Sure thing! Thanks for taking time to help.
```
ds_len: 982746
batch_size: 80
gpu_count: 8
epoch_count: 50
```

**Comment by ananyahjha93** [comment]
no worries, let me see if I can replicate this quickly

**Comment by LearnedVector** [comment]
This is fixed. there was a leftover piece of code in my lightning module where I called `self.scheduler.step(val_loss)` in the validation steps. This caused the onecycle to reset. Thanks for looking into this!

**Comment by ananyahjha93** [comment]
@LearnedVector oh ok, yeah that explains why I wasn't able to replicate this
